front/app/modules/commercial/customizable_homepage_banner/admin/CTASettings/CustomizedButtonSettings.tsx

```diff
@@ -0,0 +1,141 @@
+import { isEmpty } from 'lodash-es';
+import { Input, Label } from '@citizenlab/cl2-component-library';
+import { SectionField } from 'components/admin/Section';
+import InputMultilocWithLocaleSwitcher from 'components/UI/InputMultilocWithLocaleSwitcher';
+import React, { useMemo, useState } from 'react';
+import { CustomizedButtonConfig } from 'services/appConfiguration';
+import { CLErrors, Multiloc } from 'typings';
+import { FormattedMessage } from 'utils/cl-intl';
+import messages from '../messages';
+import genericMessages from 'components/UI/Error/messages';
+import settingsMessages from 'containers/Admin/settings/messages';
+import { injectIntl } from 'utils/cl-intl';
+import { InjectedIntlProps } from 'react-intl';
+import useAppConfigurationLocales from 'hooks/useAppConfigurationLocales';
+import { isNilOrError } from 'utils/helperUtils';
+
+// Validations for these inputs are duplicated on BE and FE.
+// We need them on BE (AppConfiguration model):
+// 1. To be sure that the values in DB are valid (for rake tasks and other things).
+// 2. To pass more or less easily the state of submit and errors from the top form component to this component.
+//    Another option would be to move validations up in the component tree, but it seems they belong here.
+//
+// We need them on FE:
+// 1. It's more user friendly.
+// 2. We use FE validations more often and so we have some translations for them on FE and not on BE (e.g. blank, wrong URL format)
+// 3. Rails model errors are simple key-value pairs, but e.g. for InputMultiloc we need messages for inputs
+//    in each language. It can be implemented, but it would be custom and complicated.
+
+// FE validations are inspired by:
+// front/app/containers/Admin/projects/edit/general/utils/validate.ts
+// front/app/containers/Admin/settings/customize/index.tsx
+
+// Similar input is used here (see validate)
+// front/app/containers/Admin/projects/edit/general/index.tsx
+const getTextErrors = (
+  textMultiloc: Multiloc | undefined,
+  errors: CLErrors,
+  formatMessage: Function,
+  tenantLocales
+) => {
+  let textError = {} as Multiloc;
```

Comment:
But don't we generally want to be more explicit?

Reply:
You could add Multiloc this way:

const textError: Multiloc = {}. 

I'd have to check if it's better than as Multiloc here (i.e. if it shouts if there are errors). But as you know, generally against type casting (`as ...`).

Reply:
Is there any difference between `const textError: Multiloc = {}` and `const textError = {} as Multiloc;`? Second also gives autocompletion.

Here they say https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions, 
that type assertions (`as`) are not used at runtime? But do `const textError: Multiloc` somehow affect runtime? I thought it affects only compilation.

Reply:
BTW changed to `const textError: Multiloc = {}`

Reply:
With `const textError: Multiloc = {}` I thought it would complain when you add a key that's not part of Locale (btw, the wrong Locale was being imported, pushed a fix). But it is not working here :/ I also thought it enforced the value type. 

I've also tried Record as explained [here](https://stackoverflow.com/questions/13315131/enforcing-the-type-of-the-indexed-members-of-a-typescript-object) in the second most upvoted answer, but couldn't get it to work. At one point, it did work when I used dot notation to assign a value (obj.a = ...) but not with bracket notation (obj[locale] = ...).

If you do as Multiloc, you say to TypeScript, "trust me, this object is a Multiloc", which in a way is correct, because all keys of a Multiloc are optional, but it's not helping us either.

I'm going to leave it as is for now. Feel free to explore further.